resources/views/frontend/needs/add.blade.php

```diff
@@ -60,6 +69,89 @@
                 </form>
             </div><!-- /.col-md-12 -->
         </div><!-- /.row -->
+        <script>
+        var map;
+        var geocoder;
+        var x;
+        var marker;
+            //initialize map using call back
+            function myMap() {
+              var myCenter = new google.maps.LatLng(6.9271,79.9912);                
+              var mapCanvas = document.getElementById("mapholder");
+              var mapOptions = {center: myCenter, zoom: 8};
+              map = new google.maps.Map(mapCanvas, mapOptions);
+              geocoder = new google.maps.Geocoder();
+              x=document.getElementById("map_messages");
+
+              //marker obj
+                marker = new google.maps.Marker({
+                    map: map,
+                    draggable: true,
+                    animation: google.maps.Animation.DROP       
+                });
+
+                //marker drag listener
+                 marker.addListener('dragend', function() {
+                    var lat = marker.getPosition().lat();
+                    var lng = marker.getPosition().lng();
+                    document.getElementById("geolocation").value=lat+","+lng;
+                    x.innerHTML=lat+","+lng;
+
+                  });
+            }
+            
+            //user input address
+            function getLocation() {
+
+                var address=document.getElementById("address").value;
+                codeAddress(address);
+            }
+            //address to longitude and latitude
+            function codeAddress(address) {
+                geocoder.geocode( { 'address': address}, function(results, status) {
+                    if (status == google.maps.GeocoderStatus.OK) {
+                        map.setCenter(results[0].geometry.location);
+                        showPosition(results[0].geometry.location);
+                    } else {
+                        showError(status);
+                    }
+                });
+            }
+
+            //show position using marker
+            function showPosition(position) {
+               
+                var lat_lng = position.lat() + "," + position.lng();
+                var latlng = new google.maps.LatLng(position.lat(), position.lng());
+                marker.setPosition(latlng);
+                document.getElementById("geolocation").value=lat_lng;
+                x.innerHTML=lat_lng;
+            }
+
+
+            function showError(error) {
+            switch(error) {
+                case "ZERO_RESULTS":
+                    x.innerHTML = "ඔබගේ ලිපිනය සොයාගත නොහැක . රතු පැහැති  සලකුණ ඇදගෙන යාමෙන්  ඔබ සිටින ස්ථානය ලකුණු කල හැක "
```

Comment:
@chamilad Do we have i18 support at the moment? Does resources/lang contain them?